app/controllers/account_activations_controller.rb

```diff
@@ -26,9 +26,9 @@ def load_user
   end
 
   def check_activation
-    return unless @user.activated? && !@user.authenticate?(:activation, params[:id])
-
-    flash[:danger] = t("layouts.messages.invalid_activation_link")
-    redirect_to root_url
+    if @user.activated? && !@user.authenticated?(:activation, params[:id])
+      flash[:danger] = t("layouts.messages.invalid_activation_link")
+      redirect_to root_url
+    end
```

Comment:
```suggestion
    return if !@user.activated? || @user.authenticate?(:activation, params[:id])
    
    flash[:danger] = t("layouts.messages.invalid_activation_link")
    redirect_to root_url
```
Vẫn dùng return if để cấu trúc gọn hơn nha